### PR TITLE
[TF APIs] Add batched matrix multiply to tensor library

### PR DESCRIPTION
**Pull request by jekbradbury**
Needed for Transformer-based models.

### CODE REVIEW

File: stdlib/public/TensorFlow/Ops.swift [hunk 1]
```diff
@@ -372,6 +372,23 @@ public extension Tensor where Scalar : Numeric {
...
 
+/// Performs batched matrix multiplication with another tensor and produces the
+/// result.
+@inlinable @inline(__always)
+@differentiable(
+  wrt: (lhs, rhs),
+  vjp: _vjpBatchedMatmul
+  where Scalar : Differentiable & FloatingPoint
+)
+public func batchedMatmul<Scalar : Numeric>(
```
Comment (rxwei) [inline stdlib/public/TensorFlow/Ops.swift]: I think `batchMatmul` is better and more familiar. It's top-level math function, so we don't have to worry too much about twisting the verb form to make it look swifty.

File: stdlib/public/TensorFlow/Ops.swift [hunk 2]
```diff
@@ -372,6 +372,23 @@ public extension Tensor where Scalar : Numeric {
...
+  _ lhs: Tensor<Scalar>,
+  _ rhs: Tensor<Scalar>,
+  adjointLHS: Bool = false,
```
Comment (rxwei) [inline stdlib/public/TensorFlow/Ops.swift]:
```suggestion
  adjointLeft: Bool = false,
```

File: stdlib/public/TensorFlow/Ops.swift [hunk 3]
```diff
@@ -372,6 +372,23 @@ public extension Tensor where Scalar : Numeric {
...
+  adjointRHS: Bool = false
```
Comment (rxwei) [inline stdlib/public/TensorFlow/Ops.swift]:
```suggestion
  adjointRight: Bool = false
```

**Review by rxwei — APPROVED** [review]

**Comment by rxwei** [comment]
I also think it's better moved to swift-apis, since we are about to move all operators out of the compiler repo at some point.

File: stdlib/public/TensorFlow/Ops.swift (hunk 1 above)
Comment (jekbradbury) [reply stdlib/public/TensorFlow/Ops.swift]: Amusingly, MKL calls this `batch_gemm` while cuBLAS calls it `gemmBatched`. I do think that `batchedMatmul` is slightly clearer (the operation is itself batched, not just the inputs) but I don't care all that much one way or the other.

File: stdlib/public/TensorFlow/Ops.swift (hunk 2 above)
Comment (jekbradbury) [reply stdlib/public/TensorFlow/Ops.swift]: Guessing you also want the unlabeled `lhs` and `rhs` arguments to become `left` and `right`?

Comment (rxwei) [reply stdlib/public/TensorFlow/Ops.swift]: Yeah. Rationale: I've never seen a precedent using `lhs`/`rhs` as argument labels, but I've seen `left`/`right`.

File: stdlib/public/TensorFlow/Ops.swift (hunk 1 above)
Comment (rxwei) [reply stdlib/public/TensorFlow/Ops.swift]: WFM.

**Pull request closed without merging**